### PR TITLE
MySQL 2 support

### LINKED ISSUE #22: Support mysql2 instrumentation
Sequelize 4 which some of our systems are using has moved from mysql to mysql2

https://github.com/sidorares/node-mysql2 
https://github.com/sequelize/sequelize/pull/6354

Its api compatible with the mysql npm package so I think to support it all you would need to do is add
`requireHook.on('mysql2', instrument);` below the following line.

https://github.com/instana/nodejs-sensor/blob/master/src/tracing/instrumentation/mysql.js#L13

### PR DESCRIPTION
**Pull request by bripkens**
@trashhalo: The tests are failing because there seems to be an API difference between mysql and mysql2. Could you take a look at this?

```
  tracing/mysql
    mysql2
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: Access denied for user ''@'172.18.0.1' (using password: NO)
    at Packet.asError (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/packets/packet.js:713:13)
    at ClientHandshake.Command.execute (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/commands/command.js:28:22)
    at Connection.handlePacket (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/connection.js:515:28)
    at PacketParser.onPacket (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/connection.js:94:16)
    at PacketParser.executeStart (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/packet_parser.js:77:14)
    at Socket.<anonymous> (/Users/ben/projects/instana/repos/nodejs-sensor/node_modules/mysql2/lib/connection.js:102:29)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
```

Fixes #22

### CODE REVIEW

File: test/tracing/mysql_test.js [hunk 1]
```diff
@@ -16,92 +16,105 @@ describe('tracing/mysql', function() {
...
-    return expressMysqlControls.addValue(42)
-    .then(function() {
-      return utils.retry(function() {
-        return agentStubControls.getSpans()
-        .then(function(spans) {
-          var entrySpan = utils.expectOneMatching(spans, function(span) {
-            expect(span.n).to.equal('node.http.server');
-            expect(span.f.e).to.equal(String(expressMysqlControls.getPid()));
-          });
+  describe.only('mysql2', function() {
```
Comment (trashhalo) [inline test/tracing/mysql_test.js]: accidentally commited .only

**Comment by trashhalo** [comment]
I don't really have bandwidth right now to look into whats up :0( Id suggest firing up a debugger to see whats not working.

**Comment by bripkens** [comment]
Works now @trashhalo. Error was caused by our prototype manipulation technique. This technique doesn't work for mysql2 as creating a connection immediately connects with mysql2.

**Comment by bripkens** [comment]
Will be reviewed & merged tomorrow during regular CET hours.

**Comment by trashhalo** [comment]
200w

**Pull request merged**